playbook/app/pb_kits/playbook/pb_timestamp/_timestamp.html.erb

```diff
@@ -1,6 +1,16 @@
 <%= content_tag(:div,
-    id: object.id,
-    data: object.data,
-    class: object.classname) do %>
-  <%= pb_rails("caption", props: { text: object.text, tag: 'span', size: 'xs' }) %>
+  aria: object.aria,
+  id: object.id,
+  data: object.data,
+  class: object.classname) do %>
+
+  <% if object.variant == "updated" || object.variant == "elapsed" %>
+    <%= pb_rails("caption", props: { text: object.format_updated_string, size: 'xs', dark: object.dark }) %>
+  <% else %>
+    <% if object.show_date %>
+      <%= pb_rails("caption", props: { text: object.timestamp ? object.format_datetime_string : object.text, size: 'xs', dark: object.dark }) %>
+    <% else %>
+      <%= pb_rails("caption", props: { text: object.format_time_string, size: 'xs', dark: object.dark }) %>
```

Comment:
What is interesting here is that both `size` and `dark` props are the same values while only the `text` prop changes based on the outer conditional. It is entirely possible to dry this part of a bit. Parsing through a lot of conditional logic in erb is feeling a little damp.

Reply:
I was thinking the same thing actually! I'll work on that